jupyterhub/traitlets.py

```diff
@@ -62,11 +62,14 @@ def validate(self, obj, value):
         If it has one of the suffixes, it is converted into the appropriate
         pure byte value.
         """
-        if isinstance(value, int):
-            return value
-        num = value[:-1]
+        if isinstance(value, float) or isinstance(value, int):
+            return float(value)
+        try:
+            num = float(value[:-1])
```

Comment:
The K8 docs often use `Mi` suffix instead of simply `M`. [ref](https://kubernetes.io/docs/tasks/configure-pod-container/assign-memory-resource/) More of an FYI than a need to necessarily change since the error handling below will look at the suffix.

Reply:
Looking at the k8 docs is an integer expected?

> The memory resource is measured in bytes. You can express memory as a plain integer or a fixed-point integer with one of these suffixes: E, P, T, G, M, K, Ei, Pi, Ti, Gi, Mi, Ki. For example, the following represent approximately the same value:
128974848, 129e6, 129M , 123Mi

Reply:
Since this isn't strictly for kubernetes, we have some freedom in what we choose. Docker, for instance, allows non-integer with a multiplier, e.g. `1.5G`.

Ultimately, though, we are always providing an integer, e.g. `1.5G` should produce the integer `15e8`. So if Kubernetes requires an int, we will still be giving it one.